### DISCUSSION TITLE
What is the difference between GeometryObject's disableCollision and activeCollisionPair?

### DISCUSSION THREAD

**Discussion by joao-pm-santos96** [stack-of-tasks/pinocchio#2239] [Q&A] [answered]
Hello all,

I am fiddling around with pinocchio's collision checking and got a question? Why are there two ways to disable the collision checking of a GeometryObject?

In here https://github.com/stack-of-tasks/pinocchio/blob/ac0b1aa6b18931bed60d0657de3c7680a4037dd3/include/pinocchio/algorithm/parallel/geometry.hpp#L32-L34 we can see that we can either disable the collision pair or set the disableCollision to False. But if I disable the collision of a given geometry obejct and then create the GeometryData associated with it, the collision pairs of that geometry remain active, which is a bit confusing... What am I missing?

Thank you!

**Comment by lmontaut** [comment, marked as answer]
Hi @joao-pm-santos96,
Depending on the state of your system, you might want to dynamically deactivate a certain collision pair (for that you can use `geom_data.deactivateCollisionPair`). 
The `disableCollision` flag allows you to do that for all collision pairs involving a specific object, so you don't have to manually check in which collision pairs this object is involved.

Simple example: there is a wall in a scene; at some point in your simulation, the wall disappears (door opening, for example).
